File: flureadium/android/src/main/kotlin/dev/mulev/flureadium/car/CarTab.kt
package dev.mulev.flureadium.car

/**
 * One root tab (Continue / Library / Search) the host's `CarContentProvider`
 * returns, decoded from its transport map. A blank id or title would render an
 * unusable row, so [fromMap] rejects those by returning null.
 */
data class CarTab(
    val id: String,
    val title: String,
    val iconName: String?,
) {
    companion object {
        /** Decodes a tab from its transport map, or null when id/title is blank. */
        fun fromMap(map: Map<*, *>): CarTab? {
            val id = (map["id"] as? String)?.takeIf { it.isNotEmpty() } ?: return null
            val title = (map["title"] as? String)?.takeIf { it.isNotEmpty() } ?: return null
            return CarTab(id = id, title = title, iconName = map["iconName"] as? String)
        }
    }
}

File: flureadium_platform_interface/lib/src/car/car_tab.dart
/// One root tab shown on a car head unit's tab bar (Continue / Library /
/// Search).
///
/// A serializable value type with runtime guards: a blank id or title would
/// render an unusable tab on the head unit, so both are rejected at
/// construction rather than in debug-only asserts.
class CarTab {
  CarTab({required this.id, required this.title, this.iconName}) {
    if (id.isEmpty) {
      throw ArgumentError.value(id, 'id', 'must not be empty');
    }
    if (title.isEmpty) {
      throw ArgumentError.value(title, 'title', 'must not be empty');
    }
  }

  /// Rebuilds a tab from its [toMap] representation.
  factory CarTab.fromMap(Map<String, Object?> map) => CarTab(
    id: map['id']! as String,
    title: map['title']! as String,
    iconName: map['iconName'] as String?,
  );

  /// Opaque host id, e.g. `library`.
  final String id;

  /// Tab label, e.g. `Library`.
  final String title;

  /// Optional platform icon hint, e.g. `books`; renderers map it to a native
  /// system image where available.
  final String? iconName;

  Map<String, Object?> toMap() => <String, Object?>{
    'id': id,
    'title': title,
    'iconName': iconName,
  };
}
